### PR TITLE
Fix Stocking Bus bug

### PR DESCRIPTION
**Pull request by miozune**
Now without ruining many people's bases.

### CODE REVIEW

**Comment by Pilad** [comment]
Stack size not working?
изображение
изображение
изображение

**Comment by Pilad** [comment]
The previous PR in the left field asked to select how many items to submit. And if I asked 3 subjects, then 3 subjects were shown on the right panel. Now the bass simply sees all objects of this type on the ME. And it doesn’t matter what stack size you set.

**Comment by lordIcocain** [comment]
stack size for pull mode

**Comment by Pilad** [comment]
> stack size for pull mode

In this mode, he sees all objects in ME and tries to send them to the bass. Why such a regime at all? And nothing changes on the left. It’s not very clear whether he really sends exactly three stacks of items or is it easy to believe?
изображение

**Comment by Pilad** [comment]
изображение
On the right I simply displayed everything that is there by the number of items.

**Comment by miozune** [comment]
The setting is for threshold of whether to pull item in auto-pull mode.
I did not touch any of actual behavior of stocking bus other than for when `endRecipeProcessing` failed extraction. What you bring here are actually the same behavior as current one.

**Comment by Pilad** [comment]
> The setting is for threshold of whether to pull item in auto-pull mode.
> I did not touch any of ME interaction behavior of stocking bus other than for when `endRecipeProcessing` failed extraction. What you bring here are actually the same behavior as current one.

Thank you. I understand.

File: src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_InputBus_ME.java [hunk 1]
```diff
@@ -445,18 +451,25 @@ public void endRecipeProcessing() {
...
+                        IAEItemStack result = sg.extractItems(request, Actionable.MODULATE, getRequestSource());
                         proxy.getEnergy()
                             .extractAEPower(request.getStackSize(), Actionable.MODULATE, PowerMultiplier.CONFIG);
                         setInventorySlotContents(i + SLOT_COUNT, oldStack);
+                        if (result == null || result.getStackSize() != toExtract) {
+                            controller.criticalStopMachine();
+                            checkRecipeResult = SimpleCheckRecipeResult
+                                .ofFailurePersistOnShutdown("stocking_bus_fail_extraction");
+                        }
                     } catch (final GridAccessException ignored) {}
```
Comment (tth05) [inline src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_InputBus_ME.java:465]: I think this exception should also cause a failure

Comment (miozune) [reply src/main/java/gregtech/common/tileentities/machines/GT_MetaTileEntity_Hatch_InputBus_ME.java:465]: Not calling `Actionable.SIMULATE` and `Actionable.MODULATE` in the same method is already weird, we choose this way to not mess up many bases.

**Review by Glease — APPROVED** [review]
looks ok

**Pull request merged**